swf/src/avm2/write.rs

```diff
@@ -124,12 +124,12 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
-    #[allow(dead_code)]
     fn write_i24(&mut self, n: i32) -> Result<()> {
-        // TODO: Verify n fits in 24-bits.
-        self.write_u8(((n >> 16) & 0xff) as u8)?;
-        self.write_u8(((n >> 8) & 0xff) as u8)?;
-        self.write_u8((n & 0xff) as u8)?;
+        let bytes = n.to_le_bytes();
```

Comment:
I'm not 100% sure about this, specifically about what happens on negative values.

Reply:
Ack, actually looking at this now, we're still all screwed up on reading/writing for s24. :-( 
 a) write_s24 is writing in the wrong endian (both before and with this PR). Can do `self.write_all(&bytes[..3]);` instead.
 b) read_s24 is still doing sign extension wrong. Only the << 16 byte should be cast to `i8` in order to sign extend correctly.

-8388600 is a good test case.

it's writing the wrong endian in the original code, too.

Reply:
Done and added tests, but I didn't verify them against Flash Player yet.